design/object-bucket/object-bucket.md

```diff
@@ -0,0 +1,164 @@
+# ObjectBucketClaim CRD and Controller
+
+## Proposed Feature
+
+**A generalized method of S3 bucket provisioning through the implementation of an ObjectBucketClaim CRD and controller.** 
+
+A cluster user will request bucket via the Kubernetes API and have returned a ConfigMap with connection information for that bucket.  
+
+One ConfigMap will be created per ObjectBucketClaim instance.  ConfigMap names will be semi-deterministic and derived from the ObjectBucketClaim name.  If an ObjectBucketClaim is created with the `generateName` field, then the name cannot be known ahead of time.
```

Comment:
Why not creating an `ObjectBucket` object? Like we do with PVC and PV.

Reply:
The K8s Pod API expects a mountable/attachable k8s primitive (configMap, secret) for data injection, so we have to use a core API resource instead of a PV analog CRD.